src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeTypeBuilder.cs

```diff
@@ -1586,33 +1586,17 @@ protected override TypeInfo CreateTypeInfoCore()
                 }
 
                 MethodAttributes methodAttrs = meth.Attributes;
-
-                // Any of these flags in the implementation flags is set, we will not attach the IL method body
-                if (((meth.GetMethodImplementationFlags() & (MethodImplAttributes.CodeTypeMask | MethodImplAttributes.PreserveSig | MethodImplAttributes.Unmanaged)) != MethodImplAttributes.IL) ||
-                    ((methodAttrs & MethodAttributes.PinvokeImpl) != (MethodAttributes)0))
-                {
-                    continue;
-                }
-
-                byte[] localSig = meth.GetLocalSignature(out int sigLength);
-
                 // Check that they haven't declared an abstract method on a non-abstract class
                 if (((methodAttrs & MethodAttributes.Abstract) != 0) && ((m_iAttr & TypeAttributes.Abstract) == 0))
                 {
                     throw new InvalidOperationException(SR.InvalidOperation_BadTypeAttributesNotAbstract);
                 }
 
                 byte[]? body = meth.GetBody();
-
-                // If this is an abstract method or an interface, we don't need to set the IL.
-
                 if ((methodAttrs & MethodAttributes.Abstract) != 0)
                 {
                     // We won't check on Interface because we can have class static initializer on interface.
                     // We will just let EE or validator to catch the problem.
-
-                    // ((m_iAttr & TypeAttributes.ClassSemanticsMask) == TypeAttributes.Interface))
-
                     if (body != null)
                         throw new InvalidOperationException(SR.Format(SR.InvalidOperation_BadMethodBody, meth.Name));
                 }
```

Comment:
` else if (body == null || body.Length == 0)` on next line can be just `if (body == null)`